### PR TITLE
[incubator/couchdb] Set COUCHDB_CLUSTER_SIZE for stateful set assembler

### PR DESCRIPTION
**Pull request by natarajaya**
#### What this PR does / why we need it:
Changes from this PR are required for couchdb-stateful-set assembler with improved peer discovery mechanics (https://github.com/kocolosk/couchdb-statefulset-assembler/pull/2).

#### Which issue this PR fixes
N/A

#### Special notes for your reviewer:
Merge after https://github.com/kocolosk/couchdb-statefulset-assembler/pull/2

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: incubator/couchdb/templates/configmap.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ metadata:
     heritage: {{ .Release.Service | quote }}
     release: {{ .Release.Name | quote }}
 data:
+  clusterSize: "{{ .Values.clusterSize }}"
```
Comment (maorfr) [inline incubator/couchdb/templates/configmap.yaml:11]: is this variable added to values.yaml and README?

Comment (natarajaya) [reply incubator/couchdb/templates/configmap.yaml:11]: Yes.

Comment (maorfr) [reply incubator/couchdb/templates/configmap.yaml:11]: Sorry for not checking myself.

Comment (natarajaya) [reply incubator/couchdb/templates/configmap.yaml:11]: No problem!

File: incubator/couchdb/values.yaml [hunk 2]
```diff
@@ -51,7 +51,7 @@ image:
 ## Sidecar that connects the individual Pods into a cluster
 helperImage:
   repository: kocolosk/couchdb-statefulset-assembler
-  tag: 1.2.0
+  tag: 1.3.0
```
Comment (kocolosk) [inline incubator/couchdb/values.yaml]: This should be `2.1.0`. I updated to `2.0.0` as part of #14925 which is why you don't see it here. Eventually we'll need to redo this work over on `stable` if that PR ever gets reviewed, but let's not hold that up.

**Review by kocolosk — APPROVED** [review]
This work is largely superseded by #15167 if you're running CouchDB 2.3 or later. I'm OK merging it, but not sure if it will be needed once that PR lands. @natarajaya , what do you think?

File: incubator/couchdb/values.yaml (hunk 2 above)
Comment (natarajaya) [reply incubator/couchdb/values.yaml]: Bumped to `2.1.0`, thanks!

**Comment by natarajaya** [comment]
@kocolosk Without changes from this PR, there is a high chance that CouchDB will end up in split brain after installing this chart (specially on GKE with Istio), so I think it makes sense to merge it even if incubator version is getting deprecated.

**Comment by kocolosk** [comment]
OK, fine by me.

**Pull request merged**